File: API/Model/ResponseModel.cs
namespace API.Model
{
    public class ResponseModel
    {
        public ResponseStatus Status { get; set; }
        public string Message { get; set; } = string.Empty;
        public object? Data { get; set; }
    }
    public class PagedResult<T>
    {
        public IReadOnlyList<T> Items { get; set; } = Array.Empty<T>();
        public int Page { get; set; }
        public int PageSize { get; set; }
        public int TotalCount { get; set; }
        public bool HasMore { get; set; }
    }

    public enum ResponseStatus
    {
        Success, Failure 
    }
}

File: UI/src/app/shared/models/response.model.ts
export class ResponseModel<T = unknown> {
    status: string | number = '';
    message: string = '';
    data: T | null = null;
}

export interface PagedResult<T> {
    items: T[];
    page: number;
    pageSize: number;
    totalCount: number;
    hasMore: boolean;
}

export function isSuccessResponse(response: Pick<ResponseModel, 'status'> | null | undefined): boolean {
    return response?.status === 0 || response?.status === '0';
}
